subprojects/gradle/kotlin-dsl-support/src/main/kotlin/com/avito/kotlin/dsl/ProjectExtensions.kt

```diff
@@ -5,56 +5,58 @@ import org.gradle.api.file.RegularFileProperty
 import java.io.File
 import kotlin.reflect.KProperty
 
-fun Project.getOptionalStringProperty(name: String): String? {
-    return if (hasProperty(name)) {
-        property(name)?.toString()
-    } else {
-        null
-    }
-}
+//TODO these methods should return Property<X>
 
-fun Project.getOptionalStringProperty(name: String, default: String): String {
-    return if (hasProperty(name)) {
-        property(name)?.toString() ?: default
+/**
+ * @param nullIfBlank there may be cases when it's ok to pass empty string to override
+ * todo true by default, false to not break anything that rely on previous behavior
+ */
+fun Project.getOptionalStringProperty(name: String, nullIfBlank: Boolean = false): String? =
```

Comment:
How about another solution for blank values?

Now we have to add additional "nullIfBlank" logic and responsibility to all optional getters. But we can compose required behavior.

`project.getOptionalStringProperty("property", nullIfBlank = true)`
vs
`project.getOptionalStringProperty("property").orNull()` 
or `.nullIfBlank()`

In the second case, the client can choose any additional logic to handle the blank value.

Reply:
i really like this idea, but i see a problem with it: i need opposite default: 
`project.getMandatoryStringProperty("property").allowBlank()` - and can't express it with property syntax, because exception must be thrown in get function

Reply:
I see. It may be possible when these functions will return a custom Property<T>